chart/Graphics/Rendering/Chart/Axis/Floating.hs

```diff
@@ -108,7 +108,7 @@ scaledAxis lap rs@(minV,maxV) ps0 = makeAxis' realToFrac realToFrac
     ps        = filter isValidNumber ps0
     range []  = (0,1)
     range _   | minV == maxV = if minV==0 then (-1,1) else
-                               let d = abs (minV * 0.01) in (minV-d,maxV+d)
+                               let d = max 1e-300 (abs (minV * 0.01)) in (minV-d,maxV+d)
               | otherwise    = rs
```

Comment:
I think 1e-300 is bad choice. Smallest normalized number 2.2250738585072014e-308 is 8 orders of magnitude away and underflow doesn't happen even for 1e-321. It's probably better to check whether maxV+d and minV-d are equal to maxV/minV. 

```
range _   | minV == maxV = case minV of
             0                            -> (-1,1)
             _ | lo /= minV && hi /= maxV -> (lo,hi)
               | otherwise                -> (minV/2,maxV*2)
          | otherwise    = rs
d  = abs (minV * 0.01)
lo = minV - d
hi = maxV + d
```